### PR TITLE
Describe how to determine authenticator attachment from transports

### PR DESCRIPTION
**Pull request by emlun**
This would merge into PR #1621.

Addresses https://github.com/w3c/webauthn/pull/1621#discussion_r709721573 . Since there wasn't a description of _how_ to determine authenticator/credential attachment from transports, this adds that as a new section similar to [§5.2.1.1. Easily accessing credential data](https://www.w3.org/TR/2021/REC-webauthn-2-20210408/#sctn-public-key-easy).

I'm submitting this as a separate (meta-) PR for visibility, because of the new section.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1670.html" title="Last updated on Oct 1, 2021, 11:18 AM UTC (183d059)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/1670/38f9f72...183d059.html" title="Last updated on Oct 1, 2021, 11:18 AM UTC (183d059)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -2739,7 +2773,7 @@ attributes.
 
         The [=[RP]=] can determine the resulting [=authenticator attachment modality|attachment=] for the created credential
         using the {{AuthenticatorAttestationResponse/getTransports()}} method of the resulting {{AuthenticatorAttestationResponse}}.
-        See [[#enum-transport]] for details.
+        See [[#sctn-attachments-from-transports]] for details.
```
Comment (sbweeden) [inline index.bs:2776]: I like the new section, but as stated in the review of #1621 I am not sure if it's deterministically possible to derive the authenticator attachment modality that was used purely from the result of `getTransports()`, specifically in the case where `getTransports()` returns more than one value, and one of the values is `internal`. The RP cannot determine which was actually used, and therefore cannot determine whether or not it is useful to prompt the end user to register the platform authenticator if one exists. Let's discuss in the WebAuthn WG call, but I'm thinking the proposal in #1168 would satisfy this goal deterministically.

**Comment by equalsJeffH** [comment]
on 2021-09-22 call:
@emlun intend to wait for PR #1668 to resolve and then revisit this PR.

File: index.bs [hunk 2]
```diff
@@ -2739,7 +2773,7 @@ attributes.
 
         The [=[RP]=] can determine the resulting [=authenticator attachment modality|attachment=] for the created credential
         using the {{AuthenticatorAttestationResponse/getTransports()}} method of the resulting {{AuthenticatorAttestationResponse}}.
```
Comment (equalsJeffH) [inline index.bs:2775]:
```suggestion
        The authenticator's [=authenticator attachment modality|attachment modality=] for the created credential
        is conveyed in {{PublicKeyCredential/authenticatorAttachment}}.
```

File: index.bs (hunk 1 above)
Comment (equalsJeffH) [reply index.bs:2776]: and delete this sentence
```suggestion
```

File: index.bs [hunk 3]
```diff
@@ -2454,6 +2454,40 @@ To remove the need to parse CBOR at all in many cases, {{AuthenticatorAttestatio
...
+
+        For example, a credential [=created on=] a mobile phone may be available as a [=platform credential=]
+        when the [=client device=] is that mobile phone,
+        and as a [=roaming credential=] when the [=client device=] is a laptop computer that can reach the mobile phone via Bluetooth.
+</dl>
+
+Note that an [=authenticator's=] [=authenticator attachment modality|attachment modality=] could change over time.
+For example, a mobile phone might at one time only support [=platform attachment=]
+but later receive updates to support [=cross-platform attachment=] as well.
+
```
Comment (equalsJeffH) [inline index.bs:2489]:
```suggestion
#### Determining Authenticator Attachment from Credential Transports #### {#sctn-attachments-from-transports}

As a part of creating a {{PublicKeyCredential}} object, as the result of a  [=registration ceremony|registration=] or [=authentication ceremony=], the [=client platform=] determines the [=authenticator attachment modality=] and reports it in {{PublicKeyCredential/authenticatorAttachment}}. This section defines how to make that determination.

If the client platform 
<dl class="switch">
    : cannot determine the {{transports}} used or the transport is not a member of {{AuthenticatorTransport}}, then
    :: the [=authenticator attachment modality=] is unknown, and the value of {{PublicKeyCredential/authenticatorAttachment}} MUST be set to null.

    : used an {{AuthenticatorTransport/internal}} transport, then
    :: {{PublicKeyCredential/authenticatorAttachment}} MUST be set to "platform".

    : used a {{transports|transport}} that is a member of {{transports}} other than {{AuthenticatorTransport/internal}},
    :: {{PublicKeyCredential/authenticatorAttachment}} MUST be set to "cross-platform".

Note: An [=authenticator's=] [=authenticator attachment modality|attachment modality=] could change over time.
For example, a mobile phone might at one time only support [=platform attachment=]
but later receive updates to support [=cross-platform attachment=] as well.
```

**Review by equalsJeffH — CHANGES_REQUESTED** [review]
As @emlun notes in the [original post](https://github.com/w3c/webauthn/pull/1670#issue-999491079), there is not "...a description of _how_ to determine authenticator/credential attachment from transports...", plus, PR #1668 obviates the need for RPs to do this determination theirselves.  However, the spec could be clearer regarding how _client platforms_ ought to determine attachment from the transport used in a request.

Here's a rough suggestion wrt how to spec the latter, HTH....

**Comment by emlun** [comment]
@equalsJeffH This all seems fair I think, but also at this point it probably makes more sense to merge this into PR #1668 instead of into #1621, given the now mutual dependency between this and #1668. I'll rebase this onto #1668 and reconfigure the PR.

**Comment by emlun** [comment]
Hm... looks like I can't do that, because #1668 is a branch in a different repository clone. Maybe just close this and migrate it over to https://github.com/z11h/webauthn instead.

**Comment by emlun** [comment]
Migrated to https://github.com/z11h/webauthn/pull/3.

File: index.bs (hunk 2 above)
Comment (emlun) [reply index.bs]: This was made obsolete by rebasing onto #1668 as the affected language no longer remains.

**Comment by kike7laguna** [comment]
@kikelaguna

**Pull request closed without merging**